### PR TITLE
ROC- 4739: Def comp/org notification

### PR DESCRIPTION
**Pull request by singh-surjeet**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-4739

### Change description ###

Notify comp/org when claimant rejects repayment plan and it gets referred to judge. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleResponse.java [hunk 1]
```diff
@@ -107,6 +107,19 @@ public PartAdmissionResponse buildWithPaymentOptionInstallments() {
                 .statementOfMeans(SampleStatementOfMeans.builder().build())
                 .build();
         }
+
+        public PartAdmissionResponse buildWithPaymentOptionInstallmentsAndParty(Party party) {
```
Comment (markry-kainos) [inline domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleResponse.java]: only 1 L in installments :)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java [hunk 2]
```diff
@@ -62,23 +63,25 @@ public ClaimantResponseService(
...
+        Claim updatedClaim = caseRepository.saveClaimantResponse(claim, claimantResponse, authorization);
         claimantResponseRule.isValid(updatedClaim);
-        formaliseResponseAcceptance(response, updatedClaim, authorization);
-        if (isRejectPartAdmitNoMediation(response, updatedClaim)) {
+        formaliseResponseAcceptance(claimantResponse, updatedClaim, authorization);
+        if (isRejectPartAdmitNoMediation(claimantResponse, updatedClaim)) {
             updateDirectionsQuestionnaireDeadline(updatedClaim, authorization);
         }
-        if (!isReferredToJudge(response)) {
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
```
Comment (markry-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: should we do this higher up? - e.g. after `claimantResponseRule.assertCanBeRequested(claim, claimantId);`

Comment (singh-surjeet) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java]: assertCanBeRequested is doing this check inside as a validation check bit differently but we wouldn't be using response until few lines later and checkstyle is likely to complain. Check `VariableDeclarationUsageDistance`

File: domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleResponse.java (hunk 1 above)
Comment (singh-surjeet) [reply domain-sample-data/src/main/java/uk/gov/hmcts/cmc/domain/models/sampledata/SampleResponse.java]: Extra L is the superstitious 'L' 😆. Rectifying it!

**Pull request merged**